internal/factsengine/gatherers/saptune.go

```diff
@@ -0,0 +1,128 @@
+}
+
+// nolint:gochecknoglobals
+var (
+	SaptuneVersionUnsupported = entities.FactGatheringError{
+		Type:    "saptune-version-not-supported",
+		Message: "currently installed version of saptune is not supported",
+	}
+
+	SaptuneUnknownArgument = entities.FactGatheringError{
+		Type:    "saptune-unknown-error",
+		Message: "the requested argument is not currently supported",
+	}
+
+	SaptuneMissingArgument = entities.FactGatheringError{
+		Type:    "saptune-missing-argument",
+		Message: "missing required argument",
+	}
+
+	SaptuneCommandError = entities.FactGatheringError{
+		Type:    "saptune-cmd-error",
+		Message: "error executing saptune command",
+	}
+
+)
+
+type SaptuneGatherer struct {
+	executor utils.CommandExecutor
+}
+
+func NewDefaultSaptuneGatherer() *SaptuneGatherer {
+	return NewSaptuneGatherer(utils.Executor{})
+}
+
+func NewSaptuneGatherer(executor utils.CommandExecutor) *SaptuneGatherer {
+	return &SaptuneGatherer{
+		executor: executor,
+	}
+}
+
+func parseJSONToFactValue(jsonStr string) (entities.FactValue, error) {
+	// Unmarshal the JSON into an interface{} type.
+	var jsonData interface{}
+	if err := json.Unmarshal([]byte(jsonStr), &jsonData); err != nil {
+		return nil, err
+	}
+
+	// Convert the parsed jsonData into a FactValue using NewFactValue.
+	return entities.NewFactValueMod(jsonData)
+}
+
+func (s *SaptuneGatherer) Gather(factsRequests []entities.FactRequest) ([]entities.Fact, error) {
+	facts := []entities.Fact{}
+	log.Infof("Starting %s facts gathering process", SaptuneGathererName)
+	saptuneRetriever, _ := saptune.NewSaptune(s.executor)
+	for _, factReq := range factsRequests {
+			var fact entities.Fact
+		if len(factReq.Argument) == 0 {
+			log.Error(SaptuneMissingArgument.Message)
+			fact = entities.NewFactGatheredWithError(factReq, &SaptuneMissingArgument)
+		} else if factValue, err := handleArgument(&saptuneRetriever, factReq.Argument); err != nil {
+			fact = entities.NewFactGatheredWithError(factReq, err)
+		} else {
+			fact = entities.NewFactGatheredWithRequest(factReq, factValue)
+		}
+		facts = append(facts, fact)
+	}
+
+	log.Infof("Requested %s facts gathered", SaptuneGathererName)
+	return facts, nil
+}
+
+func handleArgument(
+	saptuneRetriever *saptune.Saptune,
+	argument string,
+) (entities.FactValue, *entities.FactGatheringError) { 
+	internalArguments, ok := whitelistedArguments[argument]
+
+	if !ok {
+		gatheringError := SaptuneUnknownArgument.Wrap(internalArguments)
+		log.Error(gatheringError)
+		return nil, gatheringError
+	}
+
+	argList := strings.Split(internalArguments, " ")
+	saptuneOutput, commandError := saptuneRetriever.RunCommandJSON(argList...)
+	if commandError != nil {
+		gatheringError := SaptuneCommandError.Wrap(commandError.Error())
+		log.Error(gatheringError)
+		return nil, gatheringError
+	}
+
+	return gatherFactsFromOutput(saptuneOutput)
+}
+
+func gatherFactsFromOutput(commandOutput []byte) (entities.FactValue, *entities.FactGatheringError) {
```

Comment:
Do we need a new function for a `if/else` snippet?

Reply:
sure, can be removed